### DISCUSSION TITLE
Registry metadata

### DISCUSSION THREAD

**Discussion by jaytoday** [centrehq/verite#381] [Ideas]
It may be helpful for wallets, dApps and tools like the [Web3 credentials library](https://github.com/centrehq/verite/discussions/379) to be able to refer to metadata describing a verification registry, just as they can lookup metadata about an NFT at a `tokenURI` specified by an NFT contract.

This proposal will be to create a simple extension for registry [contract libraries](https://github.com/centrehq/verite/tree/main/packages/contract/contracts)  that allows a `metadataURI` to be specified for the contract.

The metadata document at the specified URI will contain a JSON object with properties. These properties will include top-level category properties and various child properties. The top level properties would likely be:

- **registry** -  Details about the registry itself, including the format and format version that would allow a wallet or dApp to infer how to interface with it. 
- **credential** -  Information about the credential associated with the registry and the issuers that can be used to obtain a credential verification. 

Other properties will be added as needed, and a more formal JSON Schema will be provided in a detailed proposal. 

Presumably, metadata will often be stored using IPFS for the purpose of immutability, or may be handled by a service or protocol assisting with providing up-to-date metadata.

**Comment by kimdhamilton** [comment]
+1, agree this would be helpful
